src/components/Chip.tsx

```diff
@@ -1,9 +1,85 @@
+export enum CHIP_VARIANTS {
+  SOLID = "bg-blue-400 dark:bg-green-400 hover:bg-blue-500 dark:hover:bg-green-500",
+  TRANSPARENT = "bg-2cg-1 hover:bg-gradient-to-r from-blue-500 to-blue-500 text-blue-500 dark:bg-2cg-3 dark:hover:bg-gradient-to-r dark:from-green-500 dark:to-green-500 dark:text-green-500",
+}
+
 interface ChipProps {
   label: string;
   onDelete?: () => void;
   className?: string;
+  variant: CHIP_VARIANTS;
+  disabled?: boolean;
 }
 
-export default function Chip({ label, onDelete, className }: ChipProps) {
-  return <></>;
+export default function Chip({
+  label,
+  onDelete,
+  className,
+  variant,
+  disabled,
+}: ChipProps) {
+  //solid chip
+  if (variant == CHIP_VARIANTS.SOLID)
```

Comment:
3 things: it's not entirely clear to me what this logic does other than display 2 svgs (that are indistinguishable without seeing the site itself). You might actually have to use a normal HTML `img`  component which will allow you to dynamically source your SVG at run time.

item 2 is a follow up: instead of if/else statements, use conditional logic statements `boolean statement ? code if true : code if false`, if you used this with strings it would looks like this:

```tsx
<img src={varient === CHIP_VARIENTS.SOLID ? 'path/to/svg1' : '/path/to/svg2'}
```

item number 3: we are using typescript, so make sure to use `===` as it also ensures the validity of types - JS is weird and some values are auto converted from string to ints to objects so forth.

Reply:
The if-else block is needed because solid and gradient Chips and Buttons are rendered differently. For the gradient ones, a reliable way to do it that I found is to just render a div with the gradient bg and then the actual button on top of it with some margins. This results in it looking like it has a gradient border. This is not necessary for the solid buttons/chips.